pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+from typing_extensions import Literal, get_args
+import collections.abc as abc
+
+import numpy as np
+import qutip
+
+
+NOISE_TYPES = Literal['doppler', 'amplitude', 'SPAM', 'dephasing']
+
+
+class SimConfig:
+    """Include additional parameters to simulation.
+
+        Keyword arguments:
+            noise (list[NOISE_TYPES]): Types of noises to be used in the
+                simulation.
+            samples_per_run (int value): Number of samples per noisy run.
+                Useful for cutting down on computing time, but unrealistic.
+            runs (int value): Number of runs needed : each run
+                draws a new random noise.
+            laser_waist (float): Waist of the gaussian laser in global pulses.
+            solver_options (qutip.Options): options for the qutip solver.
+        """
+    def __init__(self, noise: list[NOISE_TYPES] = [],
+                 runs: int = 15, samples_per_run: int = 5,
+                 temperature: float = 50., laser_waist: float = 175.,
+                 solver_options: qutip.Options = qutip.Options(max_step=5)):
+        self.noise = noise
+        self.temperature = temperature
+        self.runs = runs
+        self.samples_per_run = samples_per_run
+        self.laser_waist = laser_waist
+        self.solver_options = solver_options
+        self.init_spam()
+
+    def __str__(self) -> str:
+        s = ""
+        s += "Options:\n"
+        s += "----------       \n"
+        s += "Noise types:       " + str(self.noise) + "\n"
+        s += "Spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "Temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "Number of runs:    " + str(self.runs) + "\n"
+        s += "Samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "Laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
+        return s
```

Comment:
Just a reminder of a previous comment concerning the poor performance of string concatenation with `+` and `+=`